RELEASE.md

```diff
@@ -4,90 +4,81 @@
 
-        grep -ir '1\.2\.3-dev' *
+        grep -r '@deprecated' .
 
-3. Update ``CHANGELOG.md`` to include descriptions of the changes that made it into this release. Be sure to update the heading to include the new version (1.2.4) and the date of the release. Use the existing structure in the file as a template/guide.
+    Note any deprecated functionality that was removed in the `### Miscellaneous` section of `CHANGELOG.md`.
 
-4. Submit a pull request with these changes and let Travis run.
+4. Update ``CHANGELOG.md`` to include descriptions of all **user-facing** changes that made it into this release. Be sure to update the heading to include the new version (1.2.4) and the date of the release. Use the existing structure in the file as a template/guide.
 
-## Build the documentation
+5. Submit a pull request and merge when Travis-CI tests are passing.
 
-In the meantime, you can build the documentation and update the website.
+## Build website docs
 
-**Note:** You will need to **fully install** (including built extensions) the exact version of scikit-bio that you are editing so that Sphinx will pull docstrings from the correct version of the code. **Make sure the version of scikit-bio that is imported by ``import skbio`` is the correct one!**
+You will need to **fully install** the latest master branch of scikit-bio (including built extensions) and build the docs from this version. Make sure the version of scikit-bio that is imported by ``import skbio`` is the correct one before building the docs.
 
 1. Build the documentation locally:
 
         make -C doc clean html
 
 2. Switch to the ``gh-pages`` branch of the repository.
 
-3. Remove everything from ``docs/latest/``:
-
-        git rm -rf docs/latest/*
-
-4. Create a directory for the new version of the docs and recreate the ``latest/`` directory:
+3. Remove ``docs/latest``:
 
-        mkdir docs/1.2.4
-        mkdir docs/latest
+        git rm -rf docs/latest
 
-5. Copy over the built documentation to both ``docs/1.2.4/`` and ``docs/latest``:
+4. Copy over the built documentation to ``docs/1.2.4`` and ``docs/latest``:
 
-        cp -r <path to skbio repo>/doc/build/html/* docs/1.2.4/
-        cp -r <path to skbio repo>/doc/build/html/* docs/latest/
+        cp -r doc/build/html docs/latest
+        cp -r doc/build/html docs/1.2.4
 
-6. Add a new list item to ``index.html`` to link to ``docs/1.2.4/index.html``.
+5. Add a new list item to ``index.html`` to link to ``docs/1.2.4/index.html``.
```

Comment:
There's another step which is to copy most of the freshly rendered `README.md` into the index. In practice I did this via copying GitHub's source, but it really doesn't matter. We just probably want to leave the header alone (copy only the content).